### PR TITLE
RNMT-4515 Fixes an issue with recent devices running Android 11

### PR DESCRIPTION
**Pull request by luissilvaos**
### Description
This PR introduces the changes that fixes an issue with the recent devices running Android 11, by introducing a complex type for the representation of the SQL Handle and the SQLite operation result, instead of mixing both values into a long value.

### Context
Fixes https://outsystemsrd.atlassian.net/browse/RNMT-4515

### CODE REVIEW

File: jni/Application.mk [hunk 1]
```diff
@@ -1,5 +1,5 @@
 #APP_ABI := all
-APP_ABI := armeabi armeabi-v7a x86 x86_64 arm64-v8a
+APP_ABI := armeabi-v7a x86 x86_64 arm64-v8a
 # For future consideration (needs testing):
 #APP_ABI += mips mips64
-
+APP_PLATFORM := android-23
```
Comment (Chuckytuh) [inline jni/Application.mk:5]: Are we gaining anything from this bump ? Also, reading the docs https://developer.android.com/ndk/guides/application_mk#app_platform it states that this should match the minSdkVersion , 23 is Android 6 but we support Android 5.1 , right ? If there's no gain from this being set as we, I don't see a reason for this to be break change by bumping this to 23

File: native/sqlc.c [hunk 2]
```diff
@@ -6,32 +6,36 @@
...
 // ref: http://www.ibm.com/developerworks/opensource/tutorials/os-androidndk/index.html
 #define MYLOG(...) __android_log_print(ANDROID_LOG_VERBOSE, "sqlc", __VA_ARGS__)
 #else
 #define MYLOG(...) ;
 #endif
 
-#define HANDLE_FROM_VP(p) ( BASE_HANDLE_OFFSET + ( (unsigned char *)(p) - (unsigned char *)NULL ) )
-#define HANDLE_TO_VP(h) (void *)( (unsigned char *)NULL + (ptrdiff_t)((h) - BASE_HANDLE_OFFSET) )
+#define HANDLE_FROM_VP(p) (( (unsigned char *)(p) - (unsigned char *)NULL ) )
+#define HANDLE_TO_VP(h) (void *)( (unsigned char *)NULL + (ptrdiff_t)((h)) )
```
Comment (Chuckytuh) [inline native/sqlc.c:17]: I guess this no longer is required. Or is there something to it ?

File: jni/Application.mk (hunk 1 above)
Comment (usernuno) [reply jni/Application.mk:5]: No, we only support Android 6+ in MABS 7. Both 5 and 5.1 were dropped.

Comment (Chuckytuh) [reply jni/Application.mk:5]: So we're just introducing a breaking change without actually needing it on this package, right? What if tomorrow we need to fix something here that's needs to be applied also on MABS 5, we'll have to deal with release branches and whatnot for no reason

File: native/sqlc.c (hunk 2 above)
Comment (luissilvaos) [reply native/sqlc.c:17]: I removed the BASE_OFFSET and left it there to be an helper for the conversions `pointer -> long` and `long -> pointer`, and to minimize the changes in the code as well.

File: jni/Application.mk (hunk 1 above)
Comment (luissilvaos) [reply jni/Application.mk:5]: I understand your point, but to be honest, I don't see it as a problem. Here are my thoughts regarding this topic:
- We already have breaking changes in the `cordova-sqlite-storage` plugin for MABS 7.
- The current version used by MABS 6 has been a stable version.
- If we somehow have to deal with some fixes in the lib versions that support `cordova-sqlite-storage` for MABS 6 we can just create a branch for that.
- Also, the changes introduced in this PR aren't a simple fix that we can simply port to MABS 6. IMHO, they should only be applied to the version that will be released with MABS 7, due to the eventual impacts that they may cause.

File: java/io/liteglue/SQLiteResponse.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    public int getResult() {
+        return this.result;
+    }
+
+    public long getHandle() {
+        return this.handle;
+    }
+
+}
```
Comment (oscsx) [inline java/io/liteglue/SQLiteResponse.java:20]: Missing newline at the end

**Review by Chuckytuh — APPROVED** [review]

**Review by EiyuuZack — APPROVED** [review]
LGTM. What a headache this must have been considering upstream has not be updated in years...

File: native/SQLiteNative_JNI.c [hunk 4]
```diff
@@ -27,7 +27,14 @@ Java_io_liteglue_SQLiteNative_sqlc_1api_1db_1open__ILjava_lang_String_2I(JNIEnv
   if ( NULL != filename ) {
     (*env)->ReleaseStringUTFChars(env, filename, _strchars_filename);
   }
-  return _res;
+
+  jclass class = (*env)->FindClass(env,"io/liteglue/SQLiteResponse");
+  jmethodID constructor = (*env)->GetMethodID(env, class, "<init>", "(IJ)V");
```
Comment (oscsx) [inline native/SQLiteNative_JNI.c:32]: `"(IJ)V"`? What is this? Is it really a static thing?

File: native/SQLiteNative_JNI.c [hunk 5]
```diff
@@ -143,19 +150,26 @@ Java_io_liteglue_SQLiteNative_sqlc_1db_1open__Ljava_lang_String_2I(JNIEnv *env,
   if ( NULL != filename ) {
     (*env)->ReleaseStringUTFChars(env, filename, _strchars_filename);
   }
-  return _res;
+
+  jclass class = (*env)->FindClass(env,"io/liteglue/SQLiteResponse");
```
Comment (oscsx) [inline native/SQLiteNative_JNI.c:154]: Maybe we can refactor this code to avoid duplication?

File: native/SQLiteNative_JNI.c (hunk 4 above)
Comment (luissilvaos) [reply native/SQLiteNative_JNI.c:32]: It's the convention for the method signature. Check [this](http://journals.ecs.soton.ac.uk/java/tutorial/native1.1/implementing/method.html#:~:text=Forming%20the%20Method%20Name%20and,name%20in%20UTF%2D8%20form.).

File: native/SQLiteNative_JNI.c (hunk 5 above)
Comment (luissilvaos) [reply native/SQLiteNative_JNI.c:154]: Sure, that will be a good improvement.

File: native/SQLiteNative_JNI.c (hunk 4 above)
Comment (oscsx) [reply native/SQLiteNative_JNI.c:32]: I will never again complain about JavaScript 😂

**Comment by Chuckytuh** [comment]
Just adding context that this relates with https://github.com/storesafe/cordova-sqlite-storage/issues/954

File: native/sqlc.c (hunk 2 above)
Comment (gauthierm) [reply native/sqlc.c:17]: It's no longer required because it's no longer +number = pointer -number = error.

**Pull request merged**